File: src/main/java/dev/retrotv/utils/CommonMessageUtil.java
package dev.retrotv.utils;

import java.text.MessageFormat;
import java.util.ResourceBundle;

public class CommonMessageUtil {
    private static final ResourceBundle resourceBundle = ResourceBundle.getBundle("message");

    CommonMessageUtil() {
        throw new IllegalStateException("유틸리티 클래스 입니다.");
    }

    public static String getMessage(String key) {
        return resourceBundle.getString(key);
    }

    public static String getMessage(String key, Object... word) {
        return MessageFormat.format(resourceBundle.getString(key), word);
    }
}

File: src/main/kotlin/dev/retrotv/utils/CommonMessageUtil.kt
package dev.retrotv.utils

import java.text.MessageFormat
import java.util.*

class CommonMessageUtil private constructor() {

    init {
        throw IllegalStateException("유틸리티 클래스 입니다.")
    }

    companion object {
        private val resourceBundle = ResourceBundle.getBundle("message")
        fun getMessage(key: String): String {
            return resourceBundle.getString(key)
        }

        fun getMessage(key: String, vararg word: Any?): String {
            return MessageFormat.format(resourceBundle.getString(key), *word)
        }
    }
}
